charabia/src/normalizer/arabic.rs

```diff
@@ -1,23 +1,45 @@
 use super::{CharNormalizer, CharOrStr};
 use crate::{Script, Token};
 
-/// A global [`Normalizer`] removing the arabic Tatweel ('ـ') characters.
-/// https://www.compart.com/en/unicode/U+0640
+/// A global [`Normalizer`] for Arabic language.
+/// Arabic alphabet:ا,ب,ت,ث,ج,ح,خ,د,ذ,ر,ز,س,ش,ص,ض,ط,ظ,ع,غ,ف,ق,ك,ل,م,ن,ه,و,ي,ء
+/// Arabic text should be normalized by:
+/// - removing the arabic Tatweel ('ـ') characters.
+/// - normalizing the arabic Alef 'أ','إ','آ','ٱ' to 'ا'
+/// - normalizing the arabic Yeh 'ى' to 'ي'
+/// - Normalizing the arabic Taa Marbuta 'ة' to 'ه'
+/// https://en.wikipedia.org/wiki/Arabic_alphabet
 /// https://en.wikipedia.org/wiki/Kashida
+
 pub struct ArabicNormalizer;
 
+// All normalizers only need to implement the method `normalize_char` and the method `should_normalize` of the `CharNormalizer` trait.
 impl CharNormalizer for ArabicNormalizer {
+    // Creates the normalized version of the provided char.
     fn normalize_char(&self, c: char) -> Option<CharOrStr> {
-        (!is_tatweel(c)).then(|| c.into())
+        normalize_arabic_char(c)
     }
 
     fn should_normalize(&self, token: &Token) -> bool {
-        token.script == Script::Arabic && token.lemma().chars().any(is_tatweel)
+        token.script == Script::Arabic && token.lemma.chars().any(is_shoud_normalize)
     }
 }
 
-fn is_tatweel(c: char) -> bool {
-    c == 'ـ'
+fn normalize_arabic_char(c: char) -> Option<CharOrStr> {
+    match c {
+        'ـ' => None,
+        'أ' | 'إ' | 'آ' | 'ٱ' => Some('ا'.into()),
```

Comment:
```suggestion
'ٱ' => Some('ا'.into()),
```
We should only keep [U+0671](https://www.compart.com/en/unicode/U+0671),
The other characters should be already supported by the current implementation:
1) The [compatibility decomposition normalizer](https://github.com/meilisearch/charabia/blob/main/charabia/src/normalizer/compatibility_decomposition.rs) should decompose `آ` ([U+0622](https://www.compart.com/en/unicode/U+0622)) in [ا (U+0627)](https://www.compart.com/en/unicode/U+0627) + [◌ٓ (U+0653)](https://www.compart.com/en/unicode/U+0653)
2) The [nonspacing-mark normalizer](https://github.com/meilisearch/charabia/blob/main/charabia/src/normalizer/nonspacing_mark.rs) removes the nonspacing-marks giving [ا (U+0627)](https://www.compart.com/en/unicode/U+0627) 